sway/commands/gaps.c

```diff
@@ -0,0 +1,182 @@
+		enum gaps_op op = GAPS_OP_SET;
+		enum gaps_scope scope = GAPS_SCOPE_ALL;
+		bool inner = true;
+
+		if (strcmp(argv[0], "inner") == 0) {
+			amount_idx++;
+			inner = true;
+		} else if (strcmp(argv[0], "outer") == 0) {
+			amount_idx++;
+			inner = false;
+		}
+
+		// If one of the long variants of the gaps command is used
+		// (which starts with inner|outer) check the number of args
+		if (amount_idx > 0) { // if we've seen inner|outer
+			if (argc > 2) { // check the longest variant
+				error = checkarg(argc, "gaps", EXPECTED_EQUAL_TO, 4);
+				if (error) {
+					return error;
+				}
+			} else { // check the next longest format
+				error = checkarg(argc, "gaps", EXPECTED_EQUAL_TO, 2);
+				if (error) {
+					return error;
+				}
+			}
+		} else {
+			error = checkarg(argc, "gaps", EXPECTED_EQUAL_TO, 1);
+			if (error) {
+				return error;
+			}
+		}
+
+		if (argc == 4) {
+			// Long format: all|workspace|current.
+			if (strcmp(argv[amount_idx], "all") == 0) {
+				amount_idx++;
+				scope = GAPS_SCOPE_ALL;
+			} else if (strcmp(argv[amount_idx], "workspace") == 0) {
+				amount_idx++;
+				scope = GAPS_SCOPE_WORKSPACE;
+			} else if (strcmp(argv[amount_idx], "current") == 0) {
+				amount_idx++;
+				scope = GAPS_SCOPE_CURRENT;
+			}
+
+			// Long format: set|plus|minus
+			if (strcmp(argv[amount_idx], "set") == 0) {
+				amount_idx++;
+				op = GAPS_OP_SET;
+			} else if (strcmp(argv[amount_idx], "plus") == 0) {
+				amount_idx++;
+				op = GAPS_OP_ADD;
+			} else if (strcmp(argv[amount_idx], "minus") == 0) {
+				amount_idx++;
+				op = GAPS_OP_SUBTRACT;
+			}
+		}
+
+		double val = read_double(argv[amount_idx]);
+
+		if (isnan(val)) { // invalid <amount>
+			// guess which variant of the command was attempted
+			if (argc == 1) {
+				return cmd_results_new(CMD_INVALID, "gaps", "gaps <amount>");
+			}
+			if (argc == 2) {
+				return cmd_results_new(CMD_INVALID, "gaps",
+					"gaps inner|outer <amount>");
+			}
+			return cmd_results_new(CMD_INVALID, "gaps",
+				"gaps inner|outer all|workspace|current set|plus|minus <amount>"); 
+		}
+
+		if (amount_idx == 0) { // gaps <amount>
+			config->gaps_inner = val;
+			config->gaps_outer = val;
+			arrange_root();
+			return cmd_results_new(CMD_SUCCESS, NULL, NULL);
+		}
+		// Other variants. The middle-length variant (gaps inner|outer <amount>)
+		// just defaults the scope to "all" and defaults the op to "set".
+ 
+		double total = val;
+		if (op == GAPS_OP_SUBTRACT) {
+			total = (inner ? config->gaps_inner : config->gaps_outer) - val;
+			if (total < 0) {
+				total = 0;
+			}
+		} else if (op == GAPS_OP_ADD) {
+			total = (inner ? config->gaps_inner : config->gaps_inner) + val;
+		}
+
+		if (scope == GAPS_SCOPE_ALL) { // All: set the config values for gaps
+			if (inner) {
+				config->gaps_inner = total;
+			} else {
+				config->gaps_outer = total;
+			}
+			arrange_root();
+		} else if (scope == GAPS_SCOPE_WORKSPACE) { // Workspace: set the current workspace's gaps
```

Comment:
These comments go over the 80 column limit (It's not a hard limit, but this is significantly over).
They should go on their own line, or even on the enum definition itself.